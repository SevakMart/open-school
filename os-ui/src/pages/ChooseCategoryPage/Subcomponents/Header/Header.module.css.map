{"version":3,"sources":["Header.module.scss","../../../../styles/Mixins/FontMixin.scss","../../../../styles/Configs/Fonts.scss","../../../../styles/Configs/Colors.scss","Header.module.css"],"names":[],"mappings":"AAKI;ECJF,yBCDY;EDEZ,kBDIuC;ECHvC,gBDG8C;ECF9C,eDEkD;ECDlD,iBDCuD;EACjD,cGJY;EHKZ,sBAAA;EACA,wBAAA;EACA,YAAA;EACA,qBAAA;AIAR","file":"Header.module.css"}