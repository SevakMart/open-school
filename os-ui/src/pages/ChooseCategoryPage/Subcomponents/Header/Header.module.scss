@import "../../../../styles/Configs/Colors.scss";
@import "../../../../styles/Configs/Fonts.scss";
@import "../../../../styles/Mixins/FontMixin.scss";

.mainContainer {
    h1 {
        @include fontStyle($primary-font,normal,800,48px,59px);
        color:$header-main-color;
        box-sizing: border-box;
        margin:64px 0 48px 88px;
        width:750px;
        display:inline-block;
    }
}