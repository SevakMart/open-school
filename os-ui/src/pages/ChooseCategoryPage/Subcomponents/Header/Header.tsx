import { useTranslation } from 'react-i18next';
import { Input } from '../../../../component/Input/Input';
import styles from './Header.module.scss';

const Header = ({ changeUrlQueries }:{changeUrlQueries:(title:string)=>void}) => {
  const { t } = useTranslation();
  const { mainContainer } = styles;
  return (
    <div className={mainContainer}>
      <h1>{t('string.homePage.header.chooseCategories')}</h1>
      <Input.SearchInput
        changeUrlQueries={(title:string) => changeUrlQueries(title)}
        className={['search', 'search__subcategoriesSearch']}
        placeholder={t('Search by name')}
      />
    </div>
  );
};
export default Header;
