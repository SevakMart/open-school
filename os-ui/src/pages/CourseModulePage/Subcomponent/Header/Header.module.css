.mainContainer h1 {
  font-family: "Montserrat";
  font-style: normal;
  font-weight: 800;
  font-size: 48px;
  line-height: 59px;
  color: #222427;
  box-sizing: border-box;
  margin: 64px 0 48px 88px;
  width: 750px;
  display: inline-block;
}/*# sourceMappingURL=Header.module.css.map */