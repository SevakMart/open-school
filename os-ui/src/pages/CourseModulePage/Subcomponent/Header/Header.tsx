import NavbarOnSignIn from '../../../../component/Navbar-Component/NavbarOnSignIn/NavbarOnSignIn';
import styles from './Header.module.scss';

const Header = () => {
  const { mainContainer } = styles;
  return (
    <div className={mainContainer}>
      <NavbarOnSignIn />
    </div>
  );
};
export default Header;
