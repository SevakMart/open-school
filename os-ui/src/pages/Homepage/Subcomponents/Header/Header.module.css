.mainContent {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  margin: 0 2.7%;
  border-radius: 2.5rem;
  background-color: #F9FAFD;
  box-sizing: border-box;
}
.mainContent .leftContent {
  padding: 11% 0 11% 11%;
  width: 40%;
}
.mainContent .leftContent h1 {
  font-family: "Montserrat";
  font-style: normal;
  font-weight: 800;
  font-size: 3rem;
  line-height: 3.6rem;
  color: #222427;
}
.mainContent .leftContent p {
  font-family: "Montserrat";
  font-style: normal;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5rem;
  color: #5E617B;
}
.mainContent .leftContent .buttonContainer {
  width: 50%;
  margin-top: 8%;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
}
.mainContent .rightContent {
  width: 50%;
  text-align: center;
}/*# sourceMappingURL=Header.module.css.map */