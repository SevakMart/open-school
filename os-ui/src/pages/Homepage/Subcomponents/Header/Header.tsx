import { useTranslation } from 'react-i18next';
import { useDispatch } from 'react-redux';
import { Types } from '../../../../types/types';
import { openModal } from '../../../../redux/Slices/PortalOpenStatus';
import Navbar from '../../../../component/Navbar-Component/Navbar/Navbar';
import HomepageMainImage from '../../../../assets/svg/HomepageMainImage.svg';
import Button from '../../../../component/Button/Button';
import styles from './Header.module.scss';

const HomepageHeader = () => {
  const { t } = useTranslation();
  const dispatch = useDispatch();

  const {
    mainContent, leftContent, rightContent, buttonContainer, leftContentSignInBtn,
  } = styles;

  const handleSignUp = () => {
    dispatch(openModal({ buttonType: Types.Button.SIGN_UP }));
  };

  const handleSignIn = () => {
    dispatch(openModal({ buttonType: Types.Button.SIGN_IN }));
  };

  return (
    <>
      <Navbar />
      <div className={mainContent}>
        <div className={leftContent}>
          <h1>{t('string.homePage.header.educationalPlatform')}</h1>
          <p>{t('Join our community of mentors and help shape')}</p>
          <p>{t('the future of tech talent')}</p>
          <div className={buttonContainer}>
            <>
              <Button.SignUpButton className={['headerButton', 'headerButton__signUp']} onClick={handleSignUp}>
                {t('button.homePage.signUp')}
              </Button.SignUpButton>
              <button type="button" onClick={handleSignIn} className={leftContentSignInBtn}>
                {t('button.homePage.signIn')}
              </button>
            </>
          </div>
        </div>
        <div className={rightContent}>
          <img src={HomepageMainImage} alt="Education platform logo" />
        </div>
      </div>
    </>
  );
};
export default HomepageHeader;
