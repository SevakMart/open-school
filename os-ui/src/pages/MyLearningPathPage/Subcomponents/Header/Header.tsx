import { useEffect, useState } from 'react';
import { useTranslation } from 'react-i18next';
import { useDispatch } from 'react-redux';
import NavbarOnSignIn from '../../../../component/Navbar-Component/NavbarOnSignIn/NavbarOnSignIn';
import { getUserEnrolledCourseByCourseStatus } from '../../../../redux/Slices/UserEnrolledCourseByCourseStatus';
import styles from './Header.module.scss';

enum LearningPathNav {
    All='All',
    InProgress='InProgress',
    Completed='Completed'
}
/* eslint-disable max-len */
const Header = ({ userId, token }:{userId:number, token:string}) => {
  const { t } = useTranslation();
  const dispatch = useDispatch();
  const [activeNavType, setActiveNavType] = useState(LearningPathNav.All);
  const {
    mainHeader, courseNavigationBar, activeNav, nonActiveNav,
  } = styles;

  const handleNavigation = (e:React.SyntheticEvent) => {
    setActiveNavType((e.target as HTMLParagraphElement).dataset.testid as any);
  };

  useEffect(() => {
    switch (activeNavType) {
      case LearningPathNav.All:
        dispatch(getUserEnrolledCourseByCourseStatus({ userId, token }));
        break;
      case LearningPathNav.InProgress:
        dispatch(getUserEnrolledCourseByCourseStatus({ userId, token, params: { courseStatusId: 1 } }));
        break;
      case LearningPathNav.Completed:
        dispatch(getUserEnrolledCourseByCourseStatus({ userId, token, params: { courseStatusId: 2 } }));
        break;
    }
  }, [activeNavType]);

  return (
    <>
      <NavbarOnSignIn />
      <h1 className={mainHeader}>{t('string.myLearningPaths.mainHeaderTitle')}</h1>
      <nav className={courseNavigationBar}>
        <p className={activeNavType === 'All' ? activeNav : nonActiveNav} data-testid={LearningPathNav.All} onClick={handleNavigation}>{t('string.myLearningPaths.courseStatus.all')}</p>
        <p className={activeNavType === 'InProgress' ? activeNav : nonActiveNav} data-testid={LearningPathNav.InProgress} onClick={handleNavigation}>{t('string.myLearningPaths.courseStatus.inProgress')}</p>
        <p className={activeNavType === 'Completed' ? activeNav : nonActiveNav} data-testid={LearningPathNav.Completed} onClick={handleNavigation}>{t('string.myLearningPaths.courseStatus.completed')}</p>
      </nav>
    </>
  );
};
export default Header;
