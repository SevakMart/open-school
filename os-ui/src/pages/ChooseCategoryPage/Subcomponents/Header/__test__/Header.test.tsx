import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from '../../../../../redux/Store';
import Header from '../Header';

const mockUseNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom') as any,
  useNavigate: () => mockUseNavigate,
  useLocation: () => ({
    search: '',
  }),
}));
jest.mock('react-i18next', () => ({
  ...jest.requireActual('react-i18next'),
  useTranslation: () => ({
    t: (message:string) => {
      if (message === 'error') return 'Something went wrong please refresh the page';
    },
  }),
}));

describe('Create test case to Header component', () => {
  test('Create a snapshot test', async () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Header changeUrlQueries={jest.fn()} />
      </Provider>,
    );
    expect(asFragment).toMatchSnapshot();
  });
});
