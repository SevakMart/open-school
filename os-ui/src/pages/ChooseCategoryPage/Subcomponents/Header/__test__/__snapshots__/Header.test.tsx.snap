// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create test case to Header component Create a snapshot test 1`] = `[Function]`;
